server/eclipse-project/src/main/java/us/freeandfair/corla/controller/BallotSelection.java

```diff
@@ -134,23 +135,25 @@ public void addCvrIds(final List<Long> cvrIds) {
     }
 
     /**
-     * Return a list of CVR IDs deduped and sorted. To be given to an
-     * audit board.
+     * Return a list of CVRs, de-duplicated and sorted.
+     *
+     * The IDs of these CVRs are expected to be given to an audit board.
+     *
+     * PERF: This is not a fast operation, avoid calling it more than once.
      */
-    public List<Long> ballotSequence() {
-      return cvrs.stream()
-        .map(cvr -> cvr.id())
-        .collect(Collectors.toList());
+    public List<CastVoteRecord> cvrsInBallotSequence() {
+      return BallotSequencer.sortAndDeduplicateCVRs(
+          cvrs.stream().collect(Collectors.toList()));
     }
 
     /**
      * a good idea
      */
     public String toString() {
-      return String.format("[Segment auditSequence=%s ballotSequence=%s ballotPositions=%s ",
+      return String.format("[Segment auditSequence=%s ballotPositions=%s cvrs=%s]",
                            auditSequence(),
-                           ballotSequence(),
-                           tributes.stream().map(t -> t.ballotPosition).collect(Collectors.toList()));
+                           tributes.stream().map(t -> t.ballotPosition).collect(Collectors.toList()),
+                           this.cvrs);
```

Comment:
I don't love this but we can't pay the price of `ballotSequence()` every time we want to represent this object as a string, it'd kill performance. Showing the CVR set should be enough.